v2/block_reader.go

```diff
@@ -141,6 +142,23 @@ func (br *BlockReader) Next() (blocks.Block, error) {
 	return blocks.NewBlockWithCid(data, c)
 }
 
+// NextReader returns a CID, io.Reader and length of what should be the next block
+// the CID itself is not verified, and the reader is limited to the size of the block
+// The user of this function HAS TO consume all of the bytes in the returned reader before using any other function
+// on the BlockReader.
+// The returned length might be larger than MaxAllowedSectionSize, it is up to the user to check before loading the data into memory.
+func (br *BlockReader) NextReader() (cid.Cid, io.Reader, uint64, error) {
+	c, length, err := util.ReadNodeHeader(br.r, br.opts.ZeroLengthSectionAsEOF, math.MaxUint64)
```

Comment:
```suggestion
	c, length, err := util.ReadNodeHeader(br.r, br.opts.ZeroLengthSectionAsEOF, math.MaxInt64)
```

Or is there a reason you're using `MaxUint64` but then capping it at `MaxInt64` inside `ReadNodeHeader`? Why not just use `MaxInt64` all the way through.